others/kelvin_to_celsius.cpp

```diff
@@ -0,0 +1,73 @@
+/**
+ * @file
+ * @brief Conversion from [Kelvin to
+ * Celsius](https://byjus.com/chemistry/kelvin-to-celsius/) degrees.
+ * @details
+ * The algorithm consists on converting a Kelvin degree value to a Celsius
+ * value.
+ * The formula to convert a Kelvin to a Celsius value is:
+ * @f[ C = K - 273.15 @f] where:
+ * - C is the Celsius temperature
+ * - K is the Kelvin temperature
+ *
+ * Check out [Kelvin](https://en.wikipedia.org/wiki/Kelvin) and
+ * [Celsius](https://en.wikipedia.org/wiki/Celsius) on Wikipedia for more
+ * information about their story, how do they work, when and why they should be
+ * used, etc..
+ * @author [David Leal](https://github.com/Panquesito7)
+ */
+
+#include <cassert>   /// for assert
+#include <cmath>     /// for std::abs
+#include <iostream>  /// for IO operations
+
+/**
+ * @namespace
+ * @brief Other algorithms
+ */
+namespace others {
+/**
+ * @brief Compare two floating point numbers with a certain tolerance.
+ * This is needed as with some values, the result (e.g.: -196.15) might be a bit
+ * lower (in this case, -196.499999...).
+ * @param a the first number to compare
+ * @param b the second number to compare
+ * @param tolerance the tolerance to use when comparing the numbers
+ * @returns true if the numbers ARE equal within the given tolerance
+ * @returns false if the numbers are NOT equal within the given tolerance
+ * otherwise
+ */
+bool are_equal(double a, double b) { return std::abs(a - b) < 0.0001; }
+
+/**
+ * @brief Conversion from Kelvin to Celsius algorithm.
+ * @param number the Celsius number that will be used to convert
+ * @returns the Kelvin number converted to Celsius
+ */
+double kelvin_to_celsius(double number) { return number - 273.15; }
```

Comment:
```suggestion
double kelvin_to_celsius(double temperature_in_k) {
    const double absolute_zero_in_c = -273.15;
    if (temperature_in_k < absolute_zero_in_c) {
        throw std::invalid_argument("input temperature below absolute zero");
    }
    return temperature_in_k + absolute_zero_in_c;
}
```

Reply:
id argue temperature is enough as kelvin is already in the function name and documentation does exist to clarify such things

Reply:
I would stay with my suggestion. Variable names are read more often than the documentation. If one would like to be supper correct, I would suggest an approach like [here](https://github.com/vil02/CtUnits) (warning, the link contains some self-promotion).

Reply:
I think a clear function name precedes the variable name. Kelvin to celcius clearly implies that the parameters is most likely Kelvin. it's the age of IDEs my friend they have both code lookup as well as documention on hover. We can only handhold someone for so long. Where will we draw the line?